src/app/static/src/tests/components/downloadResults/uploadModal.test.ts

```diff
@@ -333,7 +331,46 @@ describe(`uploadModal `, () => {
         await Vue.nextTick()
         expect(wrapper.vm.$data.choiceUpload).toBe("createRelease")
         expect(mockUploadFilesToADR.mock.calls.length).toBe(2)
-        expect(mockUploadFilesToADR.mock.calls[1][mockUploadFilesToADR.mock.calls[1].length -2]["createRelease"]).toBe(true)
+        const num = mockUploadFilesToADR.mock.calls[0].length -2
+        expect(mockUploadFilesToADR.mock.calls[0][num]["uploadFiles"]).toStrictEqual([fakeMetadata["outputZip"],fakeMetadata["outputSummary"]])
+        expect(mockUploadFilesToADR.mock.calls[0][num]["createRelease"]).toBe(true)
+        expect(mockUploadFilesToADR.mock.calls[1][num]["uploadFiles"]).toStrictEqual([])
+        expect(mockUploadFilesToADR.mock.calls[1][num]["createRelease"]).toBe(true)
+        expect(mockUploadMetadataAction.mock.calls.length).toBe(2)
+    });
+
+    it(`can send upload files to ADR when download status is complete and createRelease is not flagged`, async () => {
+        const downloadResults = {
+            summary: {complete: true, downloading: false} as any,
+            spectrum: {complete: true, downloading: false} as any,
+            coarseOutput: {} as any
+        }
+        const store = createStore()
+        const wrapper = mount(UploadModal, {store})
+
+        const modal = wrapper.find(".modal");
+
+        //Refresh adrUpload upload files to trigger watch, which populates uploadFilesToADR
+        store.state.adrUpload.uploadFiles= {...fakeMetadata};
+
+        // Selects upload specific files radial
+        const radialInput = wrapper.find("#uploadFiles")
+        await radialInput.trigger("click")
+
+        //Click ok to trigger population of uploadFilesPayload
+        const okBtn = modal.find("button.btn-red");
+        expect(okBtn.element)
+        await okBtn.trigger("click");
+
+        store.state.downloadResults = downloadResults
+        await Vue.nextTick()
+        expect(wrapper.vm.$data.choiceUpload).toBe("uploadFiles")
+        expect(mockUploadFilesToADR.mock.calls.length).toBe(2)
+        const num = mockUploadFilesToADR.mock.calls[0].length -2
+        expect(mockUploadFilesToADR.mock.calls[0][num]["uploadFiles"]).toStrictEqual([fakeMetadata["outputZip"],fakeMetadata["outputSummary"]])
+        expect(mockUploadFilesToADR.mock.calls[0][num]["createRelease"]).toBe(false)
+        expect(mockUploadFilesToADR.mock.calls[1][num]["uploadFiles"]).toStrictEqual([])
```

Comment:
I was hoping that this test would be checking that the uploadFiles parameter was set to include all the files the user had left selected. Shouldn't this array be populated? And if it isn't, shouldn't the OK button be disabled?

I'm also confused about why you're expecting multiple calls to `mockUploadFilesToADR' here. Shouldn't a single click of the OK button trigger the action once, after download has completed? (Is it possible that mocks reset between tests is not happening as expected?)

Also, it would be good to include a test that the expected subset of files were sent to the action when the user unselected some before clicking OK. 

I might be making incorrect assumptions here (it's Monday morning so my brain isn't really working yet) - if so, I'd appreciate a few comments in the code to explain what's going on!

Reply:
The multiple calls to `mockUploadFilesToADR` actually predates this PR and exists in master. However, it does seem to be an artefact of testing as I see that only one call is made when I `console.log` manually. I'll look into it.